src/Identity/Endpoints/src/IdentityBearerAuthenticationHandler.cs

```diff
@@ -0,0 +1,134 @@
+        : base(optionsMonitor, loggerFactory, urlEncoder, clock)
+    {
+        _fallbackDataProtectionProvider = dataProtectionProvider;
+    }
+
+    private IDataProtectionProvider DataProtectionProvider
+        => Options.DataProtectionProvider ?? _fallbackDataProtectionProvider;
+
+    private ISecureDataFormat<AuthenticationTicket> BearerTokenProtector
+        => Options.BearerTokenProtector ?? new TicketDataFormat(DataProtectionProvider.CreateProtector(BearerTokenPurpose));
+
+    protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        // If there's no bearer token, forward to cookie auth.
+        if (await GetBearerTokenOrNullAsync() is not string token)
+        {
+            return Options.MissingBearerTokenFallbackScheme is string fallbackScheme
+                ? await Context.AuthenticateAsync(fallbackScheme)
+                : TokenMissing;
+        }
+
+        var ticket = BearerTokenProtector.Unprotect(token);
+
+        if (ticket?.Properties?.ExpiresUtc is null)
+        {
+            return FailedUnprotectingToken;
+        }
+
+        if (Clock.UtcNow >= ticket.Properties.ExpiresUtc)
+        {
+            return TokenExpired;
+        }
+
+        return AuthenticateResult.Success(ticket);
+    }
+
+    protected override async Task HandleForbiddenAsync(AuthenticationProperties properties)
+    {
+        // If there's no bearer token, forward to cookie auth.
+        if (await GetBearerTokenOrNullAsync() is null)
+        {
+            if (Options.MissingBearerTokenFallbackScheme is string fallbackScheme)
+            {
+                await Context.ForbidAsync(fallbackScheme);
+                return;
+            }
+        }
+
+        await base.HandleForbiddenAsync(properties);
+    }
+
+    protected override async Task HandleChallengeAsync(AuthenticationProperties properties)
+    {
+        // If there's no bearer token, forward to cookie auth.
+        if (await GetBearerTokenOrNullAsync() is null)
+        {
+            if (Options.MissingBearerTokenFallbackScheme is string fallbackScheme)
+            {
+                await Context.ChallengeAsync(fallbackScheme);
+                return;
+            }
+        }
+
+        Response.Headers.Append(HeaderNames.WWWAuthenticate, "Bearer");
+        await base.HandleChallengeAsync(properties);
+    }
+
+    protected override Task HandleSignInAsync(ClaimsPrincipal user, AuthenticationProperties? properties)
+    {
+        properties ??= new();
+        properties.ExpiresUtc ??= Clock.UtcNow + Options.BearerTokenExpiration;
+
+        var ticket = new AuthenticationTicket(user, properties, Scheme.Name);
+        var accessTokenResponse = new AccessTokenResponse
+        {
+            AccessToken = BearerTokenProtector.Protect(ticket),
+            ExpiresInTotalSeconds = Options.BearerTokenExpiration.TotalSeconds,
+        };
+
+        return Context.Response.WriteAsJsonAsync(accessTokenResponse);
```

Comment:
You can specify the cookie scheme if that is what you want. Otherwise sign in won't work anyway, so I don't see the harm. The alternative is just throwing, right?